src/vshaxe/display/HaxeDisplayArgumentsProvider.hx

```diff
@@ -197,13 +204,65 @@ class HaxeDisplayArgumentsProvider {
 
 		statusBarItem.hide();
 	}
+
+	function onDidChangeActiveTextEditor(editor:Null<TextEditor>) {
+		if (editor != null && editor.document.languageId == "haxe") {
```

Comment:
This `if` seems to be duplicated here unnecessarily, since `setConfigurationFromFilePath` does the same check again. Could just be `setConfigurationFromFilePath(editor!.document);` (I think we have the `!.` operator from the safety lib active in this repo).

Reply:
Looks like the safety lib is only active for the language server.